packages/syft/src/syft/client/domain_client.py

```diff
@@ -62,11 +62,24 @@ def upload_dataset(self, dataset: CreateDataset) -> Union[SyftSuccess, SyftError
                 return tuple(valid.err())
             return valid.err()
 
-    def apply_to_gateway(self, client: Self) -> None:
+    def connect_to_gateway(
+        self,
+        url: Optional[str] = None,
+        port: Optional[int] = None,
+        handle: Optional["NodeHandle"] = None,  # noqa: F821
```

Comment:
MAYBE: Instead of handle , we could have client directly ?

I think handle , is a specific , to when deployed by orchestra.

For example ,considering a domain node wanting to associate with a gateway.

1. ) They would initially explore all the gateways by

```python3
sy.gateways
```
2.) They would choose one gateway , they would like to explore

by 
```python3
gateway_client = sy.gateways[0]
```

since indexing sy.gateways automatically returns a client.

They could do directly
```python3
domain_client.connect_to_gateway(gateway_client)
```
or they would have to do
```python3
domain_client.connect_to_gateway(url = gateway_client.connection.url, port=gateway_client.connection.port)
```

Reply:
cc @IrinaMBejan

Reply:
I am going to merge this, we can address this question in a separate discussion/PR

Reply:
I see Rasswanth's point - there's no way (that I am aware of) to get a handle for a gateway when your node is deployed using hagrid as a domain server. The only way to get access to it is via client = sy.login(url="", .., ..). So when sy.orchestra is not used, how does this call look like: 

gateway_node = ???
do_client.connect_to_gateway(handle=gateway_node)

Reply:
do_client.connect_to_gateway(url="http:192.168.0.1", port=8000)

Reply:
How about we change on how its being read by the user:

`do_client.connect_to_gateway(via_client=gateway_client)`

This way we're saying that the clients are not connected but connection is made through the client.

Reply:
This would also work for me, but I do still think its kind of wasteful and potentially confusing dor users to be forced to create a client they only use for one thing (that doesnt require a client)

Reply:
oh, I like that